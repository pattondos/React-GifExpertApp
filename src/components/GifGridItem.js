import React from 'react'

const GifGridItem = ({ title, url }) => {

    return (
        <div className="cardGif animate__animated animate__bounce">
            <img
                src={url}
                alt={title}
            />
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem;
