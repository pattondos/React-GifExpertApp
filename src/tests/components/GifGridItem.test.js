import { shallow } from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en el <GifGridItem />', () => {
    const url = 'https://localhost/algo.jpg';
    const title = 'Titulo'
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar correctamente el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    });

    test('Debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe tener animate__bounce', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__bounce')).toBe(true);
    })


});
