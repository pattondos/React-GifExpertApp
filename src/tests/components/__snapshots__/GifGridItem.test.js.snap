// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="cardGif animate__animated animate__bounce"
>
  <img
    alt="Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
